test/Mono.Linker.Tests.Cases/DataFlow/MethodParametersDataFlow.cs

```diff
@@ -0,0 +1,197 @@
+﻿using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Mono.Linker.Tests.Cases.DataFlow
+{
+	[SetupCSharpCompilerToUse ("csc")]
+	[KeptMember(".ctor()")]
+	public class MethodParametersDataFlow
+	{
+		public static void Main ()
+		{
+			var instance = new MethodParametersDataFlow ();
+
+			DefaultConstructorParameter (typeof (TestType));
+			PublicConstructorsParameter (typeof (TestType));
+			ConstructorsParameter (typeof (TestType));
+			instance.InstanceMethod (typeof (TestType));
+			instance.TwoAnnotatedParameters (typeof (TestType), typeof (TestType));
+			instance.TwoAnnotatedParametersIntoOneValue(typeof (TestType), typeof (TestType));
+			instance.NoAnnotation (typeof (TestType));
+			instance.UnknownValue ();
+			instance.AnnotatedValueToUnAnnotatedParameter (typeof (TestType));
+			instance.UnknownValueToUnAnnotatedParameter ();
+			instance.UnknownValueToUnAnnotatedParameterOnInterestingMethod ();
+		}
+
+		[Kept]
+		[UnrecognizedReflectionAccessPattern (typeof (MethodParametersDataFlow), nameof (RequirePublicConstructors), new Type [] { typeof (Type) })]
+		[UnrecognizedReflectionAccessPattern (typeof (MethodParametersDataFlow), nameof (RequireConstructors), new Type [] { typeof (Type) })]
+		private static void DefaultConstructorParameter (
+			[DynamicallyAccessedMembers(DynamicallyAccessedMemberKinds.DefaultConstructor)]
+			[KeptAttributeAttribute(typeof(DynamicallyAccessedMembersAttribute))]
```

Comment:
Are you planning to hook up `DynamicallyAccessMembersAttribute` to https://github.com/mono/linker/blob/83a48f3ca59d9587669712291119ef730a6551d4/src/linker/Linker.Steps/MarkStep.cs#L515  the same way `PreserveDependencyAttribute`?

At the very least it would be nice to do from a test writing perspective so that you don't have to sprinkle `[KeptAttributeAttribute(typeof(DynamicallyAccessedMembersAttribute))]` all over the tests

Reply:
Yes - we probably should. The linker should by default remove these attributes, so that would match the desired behavior nicely.

Reply:
Filed https://github.com/mono/linker/issues/1034 to track that work